Runtimes/Supplemental/Synchronization/CMakeLists.txt

```diff
@@ -0,0 +1,122 @@
+endif()
+
+set(${PROJECT_NAME}_SWIFTC_SOURCE_DIR
+  "${PROJECT_SOURCE_DIR}/../../../"
+  CACHE FILEPATH "Path to the root source directory of the Swift compiler")
+
+set(${PROJECT_NAME}_VENDOR_MODULE_DIR "${CMAKE_SOURCE_DIR}/../cmake/modules/vendor"
+  CACHE FILEPATH "Location for private build system extension")
+
+find_package(SwiftCore QUIET REQUIRED)
+
+include(gyb)
+include(ResourceEmbedding)
+include(AvailabilityMacros)
+include(PlatformInfo)
+
+include("${${PROJECT_NAME}_VENDOR_MODULE_DIR}/Settings.cmake" OPTIONAL)
+
+option(${PROJECT_NAME}_ENABLE_LIBRARY_EVOLUTION "Generate ABI resilient runtime libraries"
+  ${SwiftCore_ENABLE_LIBRARY_EVOLUTION})
+
+option(${PROJECT_NAME}_ENABLE_PRESPECIALIZATION "Enable generic metadata prespecialization"
+  ${SwiftCore_ENABLE_PRESPECIALIZATION})
+
+option(${PROJECT_NAME}_ARCH_SUBDIR "Architecture used for setting the architecture subdirectory"
+  ${SwiftCore_ARCH_SUBDIR})
+
+option(${PROJECT_NAME}_INSTALL_NESTED_SUBDIR "Install libraries under a platform and architecture subdirectory"
+  ${SwiftCore_INSTALL_NESTED_SUBDIR})
+
+option(${PROJECT_NAME}_PLATFORM_SUBDIR "Platform name used for installed swift{doc,module,interface} files"
+  ${SwiftCore_PLATFORM_SUBDIR})
+
+option(${PROJECT_NAME}_COMPILER_VARIANT_TARGET "Compiler variant target to use when emiting modules"
+  ${SwiftCore_COMPILER_VARIANT_TARGET})
+
+include(CatalystSupport)
+include(EmitSwiftInterface)
+
+add_compile_options(
+  "$<$<COMPILE_LANGUAGE:Swift>:-nostdlibimport>"
+  "$<$<AND:$<BOOL:${${PROJECT_NAME}_ENABLE_LIBRARY_EVOLUTION}>,$<COMPILE_LANGUAGE:Swift>>:-enable-library-evolution>"
+  "$<$<AND:$<BOOL:${${PROJECT_NAME}_ENABLE_PRESPECIALIZATION}>,$<COMPILE_LANGUAGE:Swift>>:SHELL:-Xfrontend -prespecialize-generic-metadata>"
+  $<$<COMPILE_LANGUAGE:Swift>:-explicit-module-build>
+  $<$<COMPILE_LANGUAGE:Swift>:-enable-builtin-module>
+  $<$<COMPILE_LANGUAGE:Swift>:-strict-memory-safety>
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature RawLayout>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature StaticExclusiveOnly>"
+  "$<$<COMPILE_LANGUAGE:Swift>:SHELL:-enable-experimental-feature Extern>")
+
+gyb_expand(Atomics/AtomicIntegers.swift.gyb Atomics/AtomicIntegers.swift)
+gyb_expand(Atomics/AtomicStorage.swift.gyb Atomics/AtomicStorage.swift)
+
+add_library(swiftSynchronization
+  Atomics/Atomic.swift
+  Atomics/AtomicBool.swift
+  Atomics/AtomicFloats.swift
+  Atomics/AtomicLazyReference.swift
+  Atomics/AtomicMemoryOrderings.swift
+  Atomics/AtomicOptional.swift
+  Atomics/AtomicPointers.swift
+  Atomics/AtomicRepresentable.swift
+  Atomics/WordPair.swift
+  Atomics/AtomicStorage.swift
+  Atomics/AtomicIntegers.swift
+  Cell.swift
+  Mutex/Mutex.swift
+  $<$<PLATFORM_ID:Darwin>:Mutex/DarwinImpl.swift>
+  $<$<PLATFORM_ID:Linux>:Mutex/LinuxImpl.swift>
+  $<$<PLATFORM_ID:WASI>:Mutex/SpinLoopHint.swift>
+  $<$<PLATFORM_ID:WINDOWS>:Mutex/WindowsImpl.swift>)
```

Comment:
Should we not keep the `Sources` directory structure?

Reply:
The synchronization library isn't laid out this way. StringProcessing is since it's also a SwiftPM package in the separate repo. For now, let's keep the layout of whatever the library was using. We can certainly revisit the filesystem layout once we've finished the migration and are no-longer syncing the sources. Maybe worth making a note or file an issue to make things consistent though.